### PR TITLE
Add data-l10n-order to allow child elements to be ordered in DOM Overlays

### PR DESCRIPTION
**Pull request by zbraniecki**
Fixes https://github.com/projectfluent/fluent.js/issues/128

### CODE REVIEW

**Comment by Pike** [comment]
@zbraniecki, can you describe on how this is working?

I wonder if `order` is the right term here. I'm looking at this from the POV of how hard it is for localizers to not break this. Do we have to use a full blown `data-l10n-foo` attribute?

**Comment by zbraniecki** [comment]
> @zbraniecki, can you describe on how this is working?

If the `data-l10n-order` is set on the translation, we look for a matching element. If it's not, we take the first in line.

Then if the attr is set and we can't find a matching element, we take the first of the type that matches.

I think we can improve the logic, throw early if there's no match (I'd prefer the algo to be stricter and dumber initially, than to try to be smart).

> I wonder if order is the right term here.

I'm happy to bikeshed the name! I'm not convinced that `order` is the right word here, but it's not the worst. I like using `data-l10n-` prefix as it keeps all our attrs together making it more consistent.

> Do we have to use a full blown data-l10n-foo attribute?

I think so. It makes it pretty composable. Do you have an idea how to improve it?

**Comment by Pike** [comment]
First things second, I think we should do something like this. Let's figure out the details.

One thing I'd do is move the examples from "number-like" to "name-like". In the current use-case, I'd use <image data-l10n-name="addons-settings-icon"> <image data-l10n-name="extension-icon">

Which also made me pick a name just now, `name` ;-). Also matches what you used in the code, right?

As for resurrection, I'm pondering and rambling. Not an opinion yet. I guess just "randomly" assigning one element to one element would be bad, in particular if you're looking at the retranslation use-case. It'd not be idempotent for an ordering-changing locale.

Should we strip all elements of a local-name if we there are multiple, but not specified by name?

Follow-up thought, should we add  "l10n-name-extensions-icon" to the classList of the element? I guess we shouldn't assume in general that all elements of the original DOM survive throughout the lifetime of the element. That way they'd be easy to access from CSS? True dat, also works via attribute selectors.

Also should have compare-locales checks for this.

File: fluent-dom/src/overlay.js [hunk 1]
```diff
@@ -144,7 +144,7 @@ function sanitizeUsing(translationFragment, sourceElement) {
     // If a child of the same type exists in sourceElement, take its functional
     // (i.e. non-localizable) attributes. This also removes the child from
     // sourceElement.
-    const sourceChild = shiftNamedElement(sourceElement, childNode.localName);
+    const sourceChild = getNamedElement(sourceElement, childNode);
```
Comment (stasm) [inline fluent-dom/src/overlay.js]: The reason why this is called `shift...` is that it removes elements from `sourceElement`. I don't think `getNameElement` conveys this meaning well enough.

**Comment by stasm** [comment]
@zbraniecki Can we close this in favor of a new solution in line with the design discussed in #128?

**Pull request closed without merging**